### PR TITLE
feat: tpu_vm_create/get/list/delete/stop/start

### PR DESCRIPTION
**Pull request by gryczj**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [x] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork.
- [x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [x] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by subfuzion — APPROVED** [review]
LGTM, pending test issues

**Comment by m-strzelczyk** [comment]
One important note: The TPUs are a product separate from Compute (that's more politics than technical separation), but they should have their own folder. So please extract the TPU related code from the `compute` folder and make a new `tpu` folder in the root of the repo. (You'll also need to edit `CODEOWNERS` file (copy the configuration for `compute`)).

Now, as for the fate of this PR - I'm trying to get in touch with the team responsible for the TPU product and their library. Hopefully, there will be some updates to the library, maybe solving the networking issue. That's why for now, I put a `do not merge` label on this PR - let's wait and see what the team has to say. If they won't have any good news for us with reasonable timing, we'll merge this.

**Comment by gryczj** [comment]
Due to high costs of TPU Nodes, it was decided to use mocks in the tests.

**Pull request merged**
